Perpetuum.DataDumper/DataDumper.cs

```diff
@@ -0,0 +1,604 @@
+        ISparkRepository sparkRepository;
+
+        // Some static lists for helpers
+        public static SlotFlags[] sizeFlags = new SlotFlags[] { SlotFlags.small, SlotFlags.medium, SlotFlags.large };
+        public static SlotFlags[] typeFlags = new SlotFlags[] { SlotFlags.turret, SlotFlags.missile, SlotFlags.melee, SlotFlags.industrial, SlotFlags.ew_and_engineering };
+        public static SlotFlags[] locationFlags = new SlotFlags[] { SlotFlags.head, SlotFlags.chassis, SlotFlags.leg };
+        public static List<CategoryFlags> ammoWeaponCategories = new List<CategoryFlags> {
+                                                                CategoryFlags.cf_railgun_ammo,
+                                                                CategoryFlags.cf_laser_ammo,
+                                                                CategoryFlags.cf_missile_ammo,
+                                                                CategoryFlags.cf_projectile_ammo };
+
+        private static List<DataDumpConfiguration> views = new List<DataDumpConfiguration> {
+            new DataDumpConfiguration { ViewType = typeof(AmmoWeaponDataView), ViewCategory = "cf_ammo" },
+            new DataDumpConfiguration { ViewType = typeof(ModuleWeaponDataView), ViewCategory = "cf_weapons" }
+            
+        };
+
+        public DataDumper(IContainer container)
+        {
+            this.container = container;
+
+            entityFactory = container.Resolve<EntityFactory>(); 
+
+            extensionReader = container.Resolve<IExtensionReader>();
+
+            defaultReader = container.Resolve<IEntityDefaultReader>();
+
+            var productionDataReader = container.Resolve<IProductionDataAccess>();
+
+            sparkRepository = container.Resolve<ISparkRepository>();
+
+            entityServices = container.Resolve<IEntityServices>();
+
+            // var getEd = defaultReader.GetByName("def_named1_small_armor_repairer");
+            // var getCats = String.Join(";", getEd.CategoryFlags.GetCategoryFlagsTree().Where(x => x.ToString() != "undefined").Select(x => x.ToString()).ToList());
+            // var testMissile = productionDataReader.ProductionComponents[64];
+            // var testRobot = productionDataReader.ProductionComponents[193]; 
+            // var testRobot2 = productionDataReader.ProductionComponents[208];
+
+            itemNames = new Dictionary<string, string>();
+
+            var dataLines = System.IO.File.ReadAllLines(@"C:\Users\derek\Desktop\projects\Perpetuum\dictionary.tsv");
+
+            foreach (var line in dataLines)
+            {
+                var parts = line.Split(new char[] { '\t' });
+                itemNames.Add(parts[0], parts[1]);
+            }
+
+            // Now read the names from JSON files
+            var jsonFiles = System.IO.Directory.GetFiles(@"C:\Users\derek\Desktop\projects\Perpetuum\OPDB\Patches", "0.json", System.IO.SearchOption.AllDirectories);
```

Comment:
This one is tricky to pull from another repo, you should use the 0.json as-installed on the perp server folder:
`/path/to/perpetuumserver/data/customDictionary/0.json`
The server reads it from this path in the data folder, so you can too!